package UserAccounts;

enum MessageType
{
    LOGIN_REQUEST = 1;
    LOGIN_RESPONSE = 2;
    LOGOUT_REQUEST = 3;
    LOGOUT_RESPONSE = 4;
    ACCOUNT_REQUEST = 5;
    ACCOUNT_DETAILS = 6;
    UPDATE_ACCOUNT = 9;
    UPDATE_RISK = 10;
    ERROR_MESSAGE = 11;
}

enum ErrorType
{
    LOG_OFF_FAILED = 1;
    LOG_ON_FAILED = 2;
    ACCOUNT_REQUESTED_AT_LOGOFF = 5;
    ACCOUNT_NOT_LOADED = 6;
    UNKNOWN_ERROR = 7;
}

message LoginRequest
{
    required string password = 1;
}

message LoginResponse
{
    required bool isLogged = 1;
}

message LogoutResponse
{
    required bool isLogged = 1;
}

message AccountData
{
    required int32 accountid = 1;
    required string accountname = 2;
}

message RiskData
{
    required double netLiquidity = 1;
    required double netProfitLoss = 2;
    required double netMargin = 3;
    required double netBalance = 4;
    required double netEquity = 5;
}

message AccountDetails
{
    required AccountData accountdata = 1;
    required RiskData riskdata = 2;
}

message UpdateAccount
{
    required AccountData accountdata = 1;
}

message UpdateRisk
{
    required int32 accountid = 1;
    required RiskData riskdata = 2;
}

message WebMessages
{
    required MessageType type = 1;
    required string username = 2;
    optional LoginRequest loginrequest= 3;
    optional LoginResponse loginresponse = 4;
    optional LogoutResponse logoutresponse = 5;
    repeated AccountDetails accountdetails = 6;
    optional UpdateAccount updateaccount = 8;
    optional UpdateRisk updaterisk = 9;
    optional ErrorType errortype = 10;
}

